### DISCUSSION TITLE
Steps to validate webhook Saleor-Signature header

### DISCUSSION THREAD

**Discussion by tkgregory** [saleor/saleor#9822] [Q&A] [answered]
The [docs](https://docs.saleor.io/docs/3.x/developer/extending/apps/asynchronous-webhooks) provides this detail on setting up authentication for HTTPS webhooks.

> Saleor calculates the HMAC sha256 header - Saleor-Signature based on secretKey and the payload.

Is the signature based purely on the request body?

Can you please provide exact steps for validating this on the other end?

**Comment by rafalp** [comment, marked as answer]
Not exact steps, but here's an example code of how we are doing this in our Fast API boilerplate:

https://github.com/mirumee/saleor-app-framework-python/blob/905128464246f13b9fcaa4873faa8b303ea5e03e/src/saleor_app/deps.py#L97

**Reply by ashishkpaul** [reply]
@tkgregory sir facing the same problem, did you get any solution.

**Reply by tkgregory** [reply]
@ashishkpaul there are two main steps:
1. generate a signature from the request body with the secret key
2. compare that to the provided signature

You can calculate the signature with the `openssl` command line tool:
`echo -n "<request-body>" | openssl dgst -sha256 -hmac "<secret-key>"`

That's useful to make a direct request to your service with a valid `Saleor-Signature` header.

**Reply by ashishkpaul** [reply]
Thanks sir
